File: scripts/tools/copy_examples.js
const fs = require('fs');

function processFile(src, dst, depthStr) {
    let content = fs.readFileSync(src, 'utf-8');
    
    // 1. Add theme.js to head
    const headEnd = content.indexOf('</head>');
    if (headEnd !== -1) {
        const themeScript = `<script src="${depthStr}scripts/theme.js"></script>\n`;
        content = content.slice(0, headEnd) + themeScript + content.slice(headEnd);
    }
        
    // 2. Fix navigation links
    content = content.replace(/href="#"([^>]*>Home<)/g, `href="${depthStr}index.html"$1`);
    content = content.replace(/href="#"([^>]*>Courses<)/g, `href="${depthStr}pages/courses.html"$1`);
    content = content.replace(/href="#"([^>]*>Question Bank<)/g, `href="${depthStr}pages/class-12/question-bank/index.html"$1`);
    content = content.replace(/href="#"([^>]*>About<)/g, `href="${depthStr}about.html"$1`);
    content = content.replace(/href="#"([^>]*>\s*LearnNepal\s*<)/g, `href="${depthStr}index.html"$1`);
    
    // 3. Add theme-toggle class
    content = content.replace('class="material-symbols-outlined text-on-surface-variant cursor-pointer"', 'class="material-symbols-outlined text-on-surface-variant cursor-pointer theme-toggle"');
    content = content.replace('class="p-2 text-on-surface-variant hover:bg-surface-container rounded-full transition-colors"', 'class="p-2 text-on-surface-variant hover:bg-surface-container rounded-full transition-colors theme-toggle"');
    content = content.replace('class="text-on-surface-variant hover:text-primary transition-colors flex items-center justify-center p-2 rounded-full hover:bg-surface-container"', 'class="text-on-surface-variant hover:text-primary transition-colors flex items-center justify-center p-2 rounded-full hover:bg-surface-container theme-toggle"');
    
    fs.writeFileSync(dst, content, 'utf-8');
}

processFile('example/contact_us_learnnepal/code.html', 'contact.html', './');
processFile('example/courses_learnnepal/code.html', 'pages/courses.html', '../');
processFile('example/neb_question_bank/code.html', 'pages/class-12/question-bank/index.html', '../../../');
console.log('Done!');

File: scripts/tools/copy_examples.py
import re

def process_file(src, dst, depth_str):
    with open(src, 'r', encoding='utf-8') as f:
        content = f.read()
    
    # 1. Add theme.js to head
    head_end = content.find('</head>')
    if head_end != -1:
        theme_script = f'<script src="{depth_str}scripts/theme.js"></script>\n'
        content = content[:head_end] + theme_script + content[head_end:]
        
    # 2. Fix navigation links
    # Home
    content = re.sub(r'href="#"([^>]*>Home<)', rf'href="{depth_str}index.html"\1', content)
    # Courses
    content = re.sub(r'href="#"([^>]*>Courses<)', rf'href="{depth_str}pages/courses.html"\1', content)
    # Question Bank
    content = re.sub(r'href="#"([^>]*>Question Bank<)', rf'href="{depth_str}pages/class-12/question-bank/index.html"\1', content)
    # About
    content = re.sub(r'href="#"([^>]*>About<)', rf'href="{depth_str}about.html"\1', content)
    # LearnNepal Logo (some have line breaks)
    content = re.sub(r'href="#"([^>]*>\s*LearnNepal\s*<)', rf'href="{depth_str}index.html"\1', content)
    
    # 3. Add theme-toggle class
    # contact page dark mode span
    content = content.replace('class="material-symbols-outlined text-on-surface-variant cursor-pointer"', 'class="material-symbols-outlined text-on-surface-variant cursor-pointer theme-toggle"')
    # courses page dark mode button
    content = content.replace('class="p-2 text-on-surface-variant hover:bg-surface-container rounded-full transition-colors"', 'class="p-2 text-on-surface-variant hover:bg-surface-container rounded-full transition-colors theme-toggle"')
    # neb page dark mode button
    content = content.replace('class="text-on-surface-variant hover:text-primary transition-colors flex items-center justify-center p-2 rounded-full hover:bg-surface-container"', 'class="text-on-surface-variant hover:text-primary transition-colors flex items-center justify-center p-2 rounded-full hover:bg-surface-container theme-toggle"')
    
    with open(dst, 'w', encoding='utf-8') as f:
        f.write(content)

process_file('example/contact_us_learnnepal/code.html', 'contact.html', './')
process_file('example/courses_learnnepal/code.html', 'pages/courses.html', '../')
process_file('example/neb_question_bank/code.html', 'pages/class-12/question-bank/index.html', '../../../')
print('Done!')
